libs/ui/src/contest_tally.test.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { computeTallyWithPrecomputedCategories } from '@votingworks/utils';
+import { Election, ExternalTally, Tally } from '@votingworks/types';
+import {
+  electionMultiPartyPrimaryWithDataFiles,
+  electionSample2WithDataFiles,
+} from '@votingworks/fixtures';
+import { parseCvrsFileContents } from '@votingworks/test-utils';
+import { render } from '@testing-library/react';
+
+import { ContestTally } from './contest_tally';
+
+function constructTally(cvrsFileContents: string, election: Election): Tally {
+  const castVoteRecords = parseCvrsFileContents(cvrsFileContents);
+  const { overallTally } = computeTallyWithPrecomputedCategories(
+    election,
+    new Set(castVoteRecords),
+    []
+  );
+  return overallTally;
+}
+
+describe('ContestTally', () => {
```

Comment:
For the most part our tests have been using the top-level `test` without any `describe` nesting. We still have some older ones that use this pattern, but we generally haven't been creating new ones.

Reply:
Got it! I've historically liked having the ability to pair `describe`s with `beforeEach`s  for sub-conditions and corresponding setup. So I'm curious how we would feel about this "primarily flat but nested when helpful" structure:

```
beforeEach(() => {
  // Common setup
});

test(...);

test(...);

describe('When pigs fly', () => {
  beforeEach(() => {
    // Additional setup for a subset of tests
    allowPigsToFly();
  });

  test(...);

  test(...);
});
```

Reply:
Personally I prefer to either copy the shared setup into each `test` or to split the file into multiple test files.

Reply:
I didn't know about this convention and been using `describe` as well. Seems like a good topic for an eng meeting